tiny_dnn/core/framework/tensor.h

```diff
@@ -62,6 +63,61 @@
 
 namespace tiny_dnn {
 
+
+template<typename U = float_t>
+class TensorStorage {
+ public:
+    void toDevice() const {
```

Comment:
I'd keep toDevice and fromDevice private, and move the host_data, mutable_host_data, device_data and mutable_device_data into here. I think this class should be responsible for moving the data around (in the future, maybe even only part of it).